### ISSUE TITLE
[eBUG] StringIndexOutOfBoundsException when starting HO in master

### ISSUE THREAD

**Issue by tychobrailleur** [akasolace/HO#411]
**Describe the bug**
When starting HO on `master` with my existing Hattrick DB, I am getting the following exception:

```
java.lang.StringIndexOutOfBoundsException: String index out of range: 1
	at java.lang.String.substring(String.java:1963)
	at core.model.player.Player.getShortName(Player.java:1218)
	at core.gui.comp.entry.SpielerLabelEntry.createComponent(SpielerLabelEntry.java:255)
	at core.gui.comp.entry.SpielerLabelEntry.<init>(SpielerLabelEntry.java:57)
	at module.transfer.scout.TransferTableModel.initData(TransferTableModel.java:384)
	at module.transfer.scout.TransferTableModel.<init>(TransferTableModel.java:207)
	at module.transfer.scout.TransferTable.<init>(TransferTable.java:24)
	at module.transfer.scout.TransferScoutPanel.initTransferTable(TransferScoutPanel.java:221)
	at module.transfer.scout.TransferScoutPanel.initComponents(TransferScoutPanel.java:194)
	at module.transfer.scout.TransferScoutPanel.<init>(TransferScoutPanel.java:53)
	at module.transfer.TransfersPanel.initialize(TransfersPanel.java:48)
	at module.transfer.TransfersPanel.<init>(TransfersPanel.java:32)
	at module.transfer.TransfersModule.createTabPanel(TransfersModule.java:30)
	at core.gui.comp.tabbedPane.HOTabbedPane.showTab(HOTabbedPane.java:38)
	at core.gui.HOMainFrame.initComponents(HOMainFrame.java:500)
	at core.gui.HOMainFrame.<init>(HOMainFrame.java:190)
	at core.gui.HOMainFrame.instance(HOMainFrame.java:284)
	at core.HO$1.run(HO.java:225)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```
HO is not starting at all.

**To Reproduce**
1. Start HO with `HEAD` of the `master` branch (efc2d55 right now)

**Comment by tychobrailleur** [comment]
So far I have tracked the problem to a “ghost” player in `m_vScoutEintraege` in `TransferTableModel` — name is an empty string, TSI 1000, age 17, etc. As the name is empty, `core.model.player.Player#getShortName` fails at that line: https://github.com/akasolace/HO/blob/master/src/main/java/core/model/player/Player.java#L1218

I am now looking as to why I have that “ghost” player in the first place.

**Comment by tychobrailleur** [comment]
The user comes from the DB, and based on its creation date, it's been there a while:

![Screenshot 2020-03-14 at 09 24 13](https://user-images.githubusercontent.com/114285/76679126-c5408200-65d5-11ea-98a0-246bd43f6157.png)

I have checked my normal HO db, and that “ghost” player is also there.  Something must have changed to cause this...

**Comment by tychobrailleur** [comment]
a Git bisect gives this commit as the cause:

```
~/dev/HO ((6a83e89...) *|BISECTING) 🔰‏ git bisect good
46f8d2a15ec2f9332f141581fa7cb9900398cf1b is the first bad commit
commit 46f8d2a15ec2f9332f141581fa7cb9900398cf1b
Author: akasolace <thisiakasolace@gmail.com>
Date:   Thu Mar 5 21:02:31 2020 +0100

    closes #394

:040000 040000 f50b53881e95f51658069fad8a6b425fbccb3fec 474624c701f20229f0b0dd16cec9793f8363ac6d M	src

```

Having a look at that commit, it looks like `getShortName` is now being called in `SpielerLabelEntry` which is used in different place, including `TransferScoutPanel`.

A simple fix would be to not make that `substring` in `getShortName` if the player's empty.  I'll create a PR.

@akasolace I also feel `getShortName` vs. `getName` should probably be made configurable somehow, as I personally prefer to have the full name displayed.  What do you think?

**Comment by akasolace** [comment]
> @akasolace I also feel getShortName vs. getName should probably be made configurable somehow, as I personally prefer to have the full name displayed. What do you think?

In lineup tab, it was creating display issue. I decided to replicate HT interface, but indeed, it might indeed be set via preferences
